test/integration/handlers/handlers.test.js

```diff
@@ -238,8 +246,22 @@ Test('Handlers test', async handlersTest => {
         test.equal(payerCurrentPosition.value, payerExpectedPosition, 'Payer position incremented by transfer amount and updated in participantPosition')
         test.equal(payerPositionChange.value, payerCurrentPosition.value, 'Payer position change value inserted and matches the updated participantPosition value')
         test.equal(payerPositionChange.transferStateChangeId, transfer.transferStateChangeId, 'Payer position change record is bound to the corresponding transfer state change')
-        test.end()
-      }, delay)
+      }
+
+      try {
+        await retry(async bail => { // use bail(new Error('to break before max retries'))
+          const transfer = await TransferService.getById(td.messageProtocol.id) || {}
+          if (transfer.transferState !== TransferState.RESERVED) {
+            if (debug) console.log(`retrying in ${retryDelay / 1000}s..`)
+            throw new Error(`Max retry count ${retryCount} reached after ${retryCount * retryDelay / 1000}s. Tests fail`)
+          }
+          return tests()
```

Comment:
I think you hit this approach spot on. Nice!

Reply:
thanks